src/zarr/v3/sync.py

```diff
@@ -65,22 +69,19 @@ def sync(coro: Coroutine, loop: Optional[asyncio.AbstractEventLoop] = None):
             raise NotImplementedError("Calling sync() from within a running loop")
     except RuntimeError:
         pass
-    result_box: List[Optional[Any]] = [None]
-    event = threading.Event()
-    asyncio.run_coroutine_threadsafe(_runner(event, coro, result_box), loop)
-    while True:
-        # this loops allows thread to get interrupted
-        if event.wait(1):
-            break
-
-    return_result = result_box[0]
+
+    future = asyncio.run_coroutine_threadsafe(_runner(coro), loop)
+
+    done, _ = wait([future])
+    return_result = list(done)[0].result()
+
     if isinstance(return_result, BaseException):
         raise return_result
     else:
         return return_result
 
 
-def _get_loop():
+def _get_loop() -> asyncio.AbstractEventLoop | None:
```

Comment:
```suggestion
def _get_loop() -> asyncio.AbstractEventLoop:
```

I suspect you had this here because of some mypy thing but it should be possible to get this to always return a loop. Perhaps an `assert loop[0] is not None` right before the return is all that is needed?

Reply:
Might depend on whether it is called in the main thread, the IO thread of elsewhere,